RULES_DEFINITION.md

```diff
@@ -11,30 +22,238 @@ rules:
     types: ['list of types']
 ```
 
-* `key` - Required part of a rules file. A rules file must be made up of a
-          list of keys that need to be validated.
-* `required_key` - Required part of a rules file only if it is not defined in
-                   the `defaults` section.
-* `and_requires` - Optional key, determines additional keys that *must* be part
-                   of a yaml file, if a particular key is defined.
-* `or_requires` - Optional key, if a particular key is a required_key and is
-                  missing, then a yaml file may still be valid if any of the
-                  keys in the `or_requires` list exists and is properly defined.
-* `value_whitelist` - Optional key, defines a list of possible values that a key
-                      may take on. If this list is defined and a key does not
-                      have a value specified within this list then that
+### Key definitions
+
+* `defaults` - This section of the rules file is where default values can be
+               specified for the keys `required_key`, `and_requires`,
```

Comment:
This should be indented with two spaces.  It would fix your problem with bad markdown rendering.